### PR TITLE
Add a mesh_spmd for application to pass in topology

### PR DESCRIPTION
**Pull request by amithrm**

### CODE REVIEW

File: torch_xla/distributed/xla_backend.py [hunk 1]
```diff
@@ -266,6 +266,16 @@ def new_xla_process_group(ranks=None,
                           timeout=dist.default_pg_timeout,
                           backend=None,
                           pg_options=None):
+  #this options tells xla backend to "infer" a mesh
+  use_spmd = False 
+  #this option allows the application to pass in the mesh
+  mesh_spmd = None
+  if pg_options is not None and isinstance(pg_options, dict):
```
Comment (JackCaoG) [inline torch_xla/distributed/xla_backend.py:273]: can we add a test for this?

**Comment by JackCaoG** [comment]
@yeounoh do you have cycle to review this one?

**Comment by yeounoh** [comment]
> @yeounoh do you have cycle to review this one? 

Will review soon

**Review by yeounoh — APPROVED** [review]
LGTM, thank you!

**Comment by JackCaoG** [comment]
Oh I think CI failure is caused by timeout. since this pr is from a fork, it can't use the cache... let me retrigger it.

**Comment by yeounoh** [comment]
> Oh I think CI failure is caused by timeout. since this pr is from a fork, it can't use the cache... let me retrigger it.

I think the owner may also need to re-sync the fork & rebase the branch cc @amithrm

**Pull request merged**